src/Amalgam/entity/EntityQueryBuilder.h

```diff
@@ -297,7 +296,6 @@ namespace EntityQueryBuilder
 		//else don't bother parsing this, it instead contains the cases to compute case conviction for
 
 		size_t num_elements = cur_condition->positionLabels.size();
-		auto &dist_params = cur_condition->distParams;
```

Comment:
Doesn't removing this cause a bunch more derefs since you are using cur_condition->distParams a lot in changes below?